src/serena/agent.py

```diff
@@ -884,6 +1068,10 @@ def apply(
         :param body: the new symbol body. Important: Provide the correct level of indentation
             (as the original body). Note that the first line must not be indented (i.e. no leading spaces).
         """
+        if not self.ensure_language_server_for_file(relative_path):
```

Comment:
see above, instead we could `activate_language_server_for_file` and keep multiple servers alive